test/e2e/clusterctl_upgrade.go

```diff
@@ -254,6 +261,11 @@ func ClusterctlUpgradeSpec(ctx context.Context, inputGetter func() ClusterctlUpg
 		log.Logf("Creating the workload cluster with name %q using the %q template (Kubernetes %s, %d control-plane machines, %d worker machines)",
 			workLoadClusterName, "(default)", kubernetesVersion, *controlPlaneMachineCount, *workerMachineCount)
 
+		if input.PreCreateCluster != nil {
```

Comment:
I think it's confusing that this func is once called before we **wait** for cluster create but after the actual creation and then here before we actually create the cluster.

Let's move this block down below:
```go
Expect(managementClusterProxy.Apply(ctx, workloadClusterTemplate)).To(Succeed())
```